src/protocol/parser.rs

```diff
@@ -0,0 +1,130 @@
+use crate::R;
+use std::collections::VecDeque;
+use super::Command;
+
+#[derive(Debug)]
+pub(super) struct Parser {
```

Comment:
Here is where we move the `Parser` struct to. This is `pub(super)` because it should only be visible to the `src/protocol/mod.rs` and `src/protocol/command.rs` files.